shared.py

```diff
@@ -1,2 +1,43 @@
+import yaml
+import sys
+
 from slack_bolt import App
-app = App(token = open("/opt/slack-parrotbot/secrets/SLACK_BOT_TOKEN").read())
+
+if "--journald" in sys.argv:
+	from systemd import journal
+	debug_stream  = journal.stream('slack-parrotbot', priority=7)
+	info_stream   = journal.stream('slack-parrotbot', priority=6)
+	warn_stream   = journal.stream('slack-parrotbot', priority=4)
+	err_stream    = journal.stream('slack-parrotbot', priority=3)
+
+	sys.stdout = info_stream
+	sys.stderr = err_stream
+
+else:
+	debug_stream  = sys.stdout
+	info_stream   = sys.stdout
+	warn_stream   = sys.stderr
+	err_stream    = sys.stderr
+
+if "-c" in sys.argv:
+	i = sys.argv.index("-c")
+	if i < len(sys.argv) - 1:
+		config_file = sys.argv[i+1]
+	else:
+		raise Exception("-c provided but no argument follows")
+
+elif "--config" in sys.argv:
+	i = sys.argv.index("--config")
+	if i < len(sys.argv) - 1:
+		config_file = sys.argv[i+1]
+	else:
+		raise Exception("--config provided but no argument follows")
+
+else:
+	print("-c|--config not provided, defaulting to /etc/slack-parrotbot/config.yaml",
+		  file = warn_stream)
+	config_file = '/etc/slack-parrotbot/config.yaml'
```

Comment:
```suggestion
parser = argparse.ArgumentParser()
parser.add_argument('--journald', action='store_true',
					help='Use systemd journal outputs instead of stdout and stderr')
parser.add_argument('-c', '--config',
					help='Specify a path for a config.yaml',
					default='/etc/slack-parrotbot/config.yaml')
args = parser.parse_args()

if args.journald:
	from systemd import journal
	debug_stream  = journal.stream('slack-parrotbot', priority=7)
	info_stream   = journal.stream('slack-parrotbot', priority=6)
	warn_stream   = journal.stream('slack-parrotbot', priority=4)
	err_stream    = journal.stream('slack-parrotbot', priority=3)

	sys.stdout = info_stream
	sys.stderr = err_stream

else:
	debug_stream  = sys.stdout
	info_stream   = sys.stdout
	warn_stream   = sys.stderr
	err_stream    = sys.stderr

if args.config is not None:
	config_file = args.config
	if args.config == '/etc/slack-parrotbot/config.yaml':
		print("No config file specified, defaulting to /etc/slack-parrotbot-config.yaml...",
			  file = warn_stream)

```

Adds bulk of argument parsing via argparse and simplifies the code a bit

Reply:
When would `args.config` be `None`?

Reply:
Ideally never due to the specified default, but I'm paranoid with this stuff

Reply:
ok, let's add a "this should never happen" warning as well just to avoid confusion